### PR TITLE
chore: publish docs artifact

### PR DESCRIPTION
**Pull request by ttypic**
Added action to publish autogenerated docs

## Summary by CodeRabbit

- **New Features**
	- Introduced a new workflow for automated API documentation generation and deployment.
	- Added Dokka plugin for enhanced documentation capabilities.

- **Documentation**
	- Updated plugin declarations to include Dokka for documentation generation.

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        uses: gradle/gradle-build-action@v2
+
+      - name: Build Documentation
+        run: ./gradlew dokkaHtml
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-region: eu-west-2
+          role-to-assume: arn:aws:iam::${{ secrets.ABLY_AWS_ACCOUNT_ID_SDK }}:role/ably-sdk-builds-ably-chat-js
```
Comment (AndyTWF) [inline .github/workflows/docs.yml:39]: Should we have a new role for Kotlin?

Comment (ttypic) [reply .github/workflows/docs.yml:39]: Sure, will do, waiting for permissions to create new IAM role  :)

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        uses: gradle/gradle-build-action@v2
+
+      - name: Build Documentation
+        run: ./gradlew dokkaHtml
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-region: eu-west-2
+          role-to-assume: arn:aws:iam::${{ secrets.ABLY_AWS_ACCOUNT_ID_SDK }}:role/ably-sdk-builds-ably-chat-koltin
```
Comment (AndyTWF) [inline .github/workflows/docs.yml:39]:
```suggestion
          role-to-assume: arn:aws:iam::${{ secrets.ABLY_AWS_ACCOUNT_ID_SDK }}:role/ably-sdk-builds-ably-chat-kotlin
```

Typo?

**Review by sacOO7 — COMMENTED** [review]
Is the github action CI in process or hanged? 
Since yesterday, it's showing `checking for ability to merge automatically` or 
`Processing updated` : (

**Review by AndyTWF — APPROVED** [review]

**Review by sacOO7 — APPROVED** [review]
LGTM

**Pull request merged**